builtins/web/streams/native-stream-source.h

```diff
@@ -51,7 +51,8 @@ class NativeStreamSource : public BuiltinNoConstructor<NativeStreamSource> {
   static bool pull(JSContext *cx, unsigned argc, JS::Value *vp);
   static bool cancel(JSContext *cx, unsigned argc, JS::Value *vp);
   static JSObject *create(JSContext *cx, JS::HandleObject owner, JS::HandleValue startPromise,
-                          PullAlgorithmImplementation *pull, CancelAlgorithmImplementation *cancel);
+                          PullAlgorithmImplementation *pull, CancelAlgorithmImplementation *cancel,
+                          JS::HandleFunction size = nullptr, double highWaterMark = 0.0);
```

Comment:
The comment in `RequestOrResponse::create_body_stream` was confusing to me until I realized that not passing a HWM uses an `0.0` default. I think in this case it'd be clearer to make that explicit and not have a default value here at all.
```suggestion
                          JS::HandleFunction size = nullptr, double highWaterMark = 0.0);
```
```suggestion
                          JS::HandleFunction size = nullptr, double highWaterMark);
```

Reply:
I would think `0.0`  is still a sensible default considering all builtin code sets `HWM` to that value. Maybe we can move the comment to the `native-stream` implementation that would motivate why we choose to overwrite the default value from `1.0` to `0.0`.

I'm fine with either, although I would prefer having argument list in this function somewhat similar to `NewReadableStreamDefaultObject`. If we remove default value for `HWM` I guess we also need to change the order of `size` and `HWM` as the compiler won't let us having default argument followed by argument with explicit value.

Reply:
ah, good point about keeping things similar to `NewReadableStreamDefaultObject`.

I think I agree that moving the comment to `native-stream` makes sense. You'd then have to adjust it to not talk about http bodies anymore, and instead focus on explaining why for native streams a different default is chosen. But that seems like a good change to make

Reply:
OK the comment is now in the `natvie-stream` header file.